buffer/src/main/java/io/netty/buffer/ReadOnlyAbstractByteBuf.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+/**
+ * Specialized {@link ReadOnlyByteBuf} sub-type which allows fast access to parent
+ * without extra bound-checks.
+ */
+final class ReadOnlyAbstractByteBuf extends ReadOnlyByteBuf {
```

Comment:
My only complaint is that the `ReadOnlyAbstractByteBuf` name is pretty grating when this 1) extends `ReadOnlyByteBuf`, and 2) is `final`. 😅

Reply:
It wraps an AbstractByteBuf tho ? 🤷‍♂️

Reply:
Yeah, I know. I can't think of a better name.